degica/komoju-woocommerce
# Editable KOMOJU endpoint

## Context

This PR allows plugin users to change the KOMOJU endpoint used for API calls.

![image](https://user-images.githubusercontent.com/2793160/171773888-b84594e7-a627-4511-8ee5-59c44baefd4e.png)

This allows a handful of handy use-cases:
1. We can end-to-end test against our Staging environment
2. We can provide special per-merchant KOMOJU endpoints for high-traffic sites
3. Plugin users can stub out KOMOJU for their own internal testing

## How to test

(initial setup - skip if you already have the plugin up and running)
1. `docker-compose up`
2. `cd tests && npx cypress run` :point_left: the Cypress tests will install and configure the plugin for you
(testing this feature)
1. Visit the WooCommerce settings page
2. Click the KOMOJU tab
3. See "KOMOJU Endpoint" field on the bottom of the form
4. Change it to something else
5. You'll likely see an "unable to reach KOMOJU" message - this is normal if your new endpoint is not actually KOMOJU!
6. Click "reset" to change it back
